### ISSUE TITLE
Encoding Messages with a 0-length DLC?

### ISSUE THREAD

**Issue by karlding** [eerimoq/cantools#126]
I have a message that has a `DLC` of `0`, which seems to be valid according to the CAN specification (since the message DLC can be 0-8 bytes). However, `encode_data` currently does not appear to handle this case, and raises a `ValueError` instead.

```
ValueError: invalid literal for int() with base 16: b''
```

Just to make sure this is an actual issue (and not me being silly), is this how you would encode a message with a 0-length DLC?

```python
import cantools

if __name__ == '__main__':
    db = cantools.db.Database()

    db.nodes.append(cantools.db.Node('FakeNode', 'Comment'))

    signals = []
    message = cantools.db.Message(0x20, 'ZeroLength', 0, signals, senders=['FakeNode'])
    db.messages.append(message)

    # Load
    new_db = cantools.database.load_string(db.as_dbc_string())
    message = new_db.get_message_by_name('ZeroLength')

    # Empty dict -> no data
    data = message.encode({})
```

I see that `encode_data` is currently implemented like so:

```python
def encode_data(data, fields, formats, scaling):
    unpacked = {
        field.name: _encode_field(field, data, scaling)
        for field in fields
    }
    big_packed = formats.big_endian.pack(unpacked)
    little_packed = formats.little_endian.pack(unpacked)[::-1]
    packed_union = int(binascii.hexlify(big_packed), 16)
    packed_union |= int(binascii.hexlify(little_packed), 16)

    return packed_union
```

**Comment by eerimoq** [comment]
Hello!

It's a bug. I implemented a fix in version 32.13.1, available on PyPI.

**Comment by karlding** [comment]
Thanks!
